lib/server.js

```diff
@@ -281,8 +281,8 @@ function sendErrorMessage (req, res, code) {
  * @api public
  */
 
-Server.prototype.generateId = function (req) {
-  return base64id.generateId();
+Server.prototype.generateId = function (req, callback) {
+  callback(base64id.generateId());
```

Comment:
How about `callback(null, base64id.generateId());`, so that errors can also be handled (see https://github.com/socketio/engine.io/pull/518)?

Reply:
You're right! I'll update the code.